### PR TITLE
Add GRPC agent

### PR DESCRIPTION
**Pull request by ricardo6142dh**
This PR implements gRPC agents to replace the existing HTTP-based agents, aiming to improve performance and efficiency. gRPC offers lower latency and better resource utilization due to its use of HTTP/2 and Protocol Buffers. Code has been refactored to handle gRPC communication, including updated error handling and response processing.

### CODE REVIEW

**Comment by fornellas** [comment]
@ricardo6142dh I think this file:

image

must be added to the [rrb ignore pattern](https://github.com/fornellas/resonance/pull/145/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R249), otherwise, it can't finish the build, ever, as the file changes, rrb kills the process, the file changes...

rrb is linux-only, so this is why you didn't catch this in Darwin.

**Comment by ricardo6142dh** [comment]
> @ricardo6142dh I think this file:
> 
> image
> 
> must be added to the [rrb ignore pattern](https://github.com/fornellas/resonance/pull/145/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R249), otherwise, it can't finish the build, ever, as the file changes, rrb kills the process, the file changes...
> 
> rrb is linux-only, so this is why you didn't catch this in Darwin.

fixed!

**Review by wguto — APPROVED** [review]
LGTM

**Comment by fornellas** [comment]
No merge with unresolved comments plz.

**Review by fornellas — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -237,7 +238,7 @@ else
 GO_BUILD_AGENT_GOARCHS := $(GOARCH)
 endif
 
-GO_BUILD_MAX_AGENT_SIZE := 5242880
+GO_BUILD_MAX_AGENT_SIZE := 52428800
```
[the remark this answers is not in the document]
Comment (ricardo6142dh) [reply Makefile]: Fixed

File: internal/host/agent_server_http/net/net.go [hunk 2]
```diff
@@ -70,3 +70,35 @@ func (c Conn) SetReadDeadline(t time.Time) error {
 func (c Conn) SetWriteDeadline(t time.Time) error {
 	return os.ErrNoDeadline
 }
+
+type CustomListener struct {
```
[the remark this answers is not in the document]
Comment (ricardo6142dh) [reply internal/host/agent_server_http/net/net.go]: Fixed

File: internal/host/agent_grpc_client.go [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+	if err != nil {
+		return err
+	}
+
+	stdoutReader, stdoutWriter, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+
+	a.Client, err = grpc.NewClient("127.0.0.1",
```
[the remark this answers is not in the document]
Comment (ricardo6142dh) [reply internal/host/agent_grpc_client.go:126]: Fixed

**Pull request merged**